test/test_nestedtensor.py

```diff
@@ -96,8 +98,66 @@ def random_nt(device, dtype, num_tensors, max_dims, min_dims=None):
         ts1.append(t1)
     return torch.nested.nested_tensor(ts1, device=device, dtype=dtype)
 
+def define_crossref_xfails():
+    # These all fail because they call _nested_tensor_from_tensor_list(), which
+    # isn't supported.
+    crossref_xfails = [
+        "TestNestedTensor.test_2d*",
+        "TestNestedTensor.test_3d*",
+        "TestNestedTensor.test_copy_",
+        "TestNestedTensor.test_default_nested_tensor",
+        "TestNestedTensor.test_dim",
+        "TestNestedTensor.test_fill_",
+        "TestNestedTensor.test_is_contiguous",
+        "TestNestedTensor.test_like_functions*",
+        "TestNestedTensor.test_nested_namespace",
+        "TestNestedTensor.test_nested_tensor_matching_dim",
+        "TestNestedTensor.test_numel",
+        "TestNestedTensor.test_repr_string",
+        "TestNestedTensor.test_size",
+        "TestNestedTensor.test_size_dim",
+        "TestNestedTensor.test_stride",
+        "TestNestedTensor.test_to",
+        "TestNestedTensor.test_to_padded_tensor_on_empty_tensor",
+        "TestNestedTensor.test_unbind_*",
+        "TestNestedTensor.test_unbind_dim",
+        "TestNestedTensor.test_zero_",
+    ]
+    expectedFailureIfCrossRef = expectedFailureIf(TEST_WITH_CROSSREF)
+    for full_test_name in crossref_xfails:
+        test_cls_name, test_name = full_test_name.split('.')
+        test_cls = globals().get(test_cls_name)
+        if test_name.endswith('*'):
+            # Decorate all tests that match the prefix
+            for cls_attr in set(test_cls.__dict__.keys()):
```

Comment:
Nit Q: do you need the set wrapper I thought keys are itterable and unique?

Reply:
True, not sure how that ended up there. Fixed